tests/callbacks/test_model_checkpoint.py

```diff
@@ -0,0 +1,101 @@
+import os
+import pickle
+import platform
+from pathlib import Path
+
+import pytest
+
+import tests.base.develop_utils as tutils
+from pytorch_lightning import Trainer
+from pytorch_lightning.callbacks import ModelCheckpoint
+from pytorch_lightning.loggers import TensorBoardLogger
+from tests.base import EvalModelTemplate
+
+
+@pytest.mark.parametrize('save_top_k', [-1, 0, 1, 2])
+def test_model_checkpoint_with_non_string_input(tmpdir, save_top_k):
+    """ Test that None in checkpoint callback is valid and that chkp_path is set correctly """
+    tutils.reset_seed()
+    model = EvalModelTemplate()
+
+    checkpoint = ModelCheckpoint(filepath=None, save_top_k=save_top_k)
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        checkpoint_callback=checkpoint,
+        overfit_pct=0.20,
+        max_epochs=5,
+    )
+    trainer.fit(model)
+
+    # These should be different if the dirpath has be overridden
+    assert trainer.ckpt_path != trainer.default_root_dir
```

Comment:
shall we also check how many chpts were created?

Reply:
I guess the goal of this test about the paths. But there is already another test that checks what you mean, it is called "test_model_checkpoint_options"